### ISSUE TITLE
[Super AI] [Good First Issue] Convert GPTJ Self Attention to XFormer

### ISSUE THREAD

**Issue by new5558** [OpenThaiGPT/openthaigpt-pretraining#73]
Description:
[GPTJ](https://huggingface.co/EleutherAI/gpt-j-6b) is one of the [most popular](https://huggingface.co/models?pipeline_tag=text-generation) Open Sources of GPT model available on the Internet. However, the current Huggingface [code](https://github.com/huggingface/transformers/blob/main/src/transformers/models/gptj/modeling_gptj.py) of GPTJ is using the old Attention mechanism. In this task, we want to convert the original code from GPTJ to support Flash Attetnion by doing

1. Download the model from Huggingface and try to run inference using their official implementation code. (6.7B Version is enough)
2. Convert to both:
- Pytorch 2.0 Attention: https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html
- XFormer : https://github.com/HazyResearch/flash-attention
3. Try to run inference again and get the same result (logits)

Deliverables:
- Convert/Inference Code Script
- PR
- Unit Test to make sure converted model have similar output to the original one
